Lib/tkinter/__init__.py

```diff
@@ -763,15 +763,31 @@ def after_cancel(self, id):
 
         Identifier returned by after or after_idle must be
         given as first parameter."""
+        if id is None:
+            return
         try:
-            data = self.tk.call('after', 'info', id)
-            # In Tk 8.3, splitlist returns: (script, type)
-            # In Tk 8.4, splitlist may return (script, type) or (script,)
+            data = self.after_info(id)
             script = self.tk.splitlist(data)[0]
             self.deletecommand(script)
         except TclError:
             pass
         self.tk.call('after', 'cancel', id)
+
+    def after_info(self, id=None):
+        """Return information about existing event handlers.
+
+        With no argument, return a tuple of the identifiers for all existing
+        event handlers created by the after and after_idle commands for this
+        interpreter.  If id is supplied, it specifies an existing handler; id
+        must have been the return value from some previous call to after or
+        after_idle and it must not have triggered yet or been canceled. If the
+        id doesn't exist, a TclError is raised.  Othewise, the return value is
+        a tuple containing (script, type) where script is a reference to the
+        function to be called by the event handler and type is either 'idle'
+        or 'timer' to indicate what kind of event handler it is.
+        """
+        return self.tk.splitlist(self.tk.call('after', 'info', id))
```

Comment:
As I say elsewhere, I would want the return to be (python_function, type).  I expect that a bad or non-existent id would raise TclError.  If so, TclError should be caught, as in after_cancel, but here replaced, perhaps by ValueError, instead of ignored.

Reply:
I think TclError is good here. I have doubts about the result type (see more details on the tracker).

Reply:
Read it.  Something easy is better than either nothing or something awkward to impossible.